api/transport/header.go

```diff
@@ -38,6 +38,8 @@ func CanonicalizeHeaderKey(k string) string {
 type Headers struct {
 	// This representation allows us to make zero-value valid
 	items map[string]string
+	// map from canonicalized key to original key
+	keyMapping map[string]string
```

Comment:
nit: "*mapping" for a map is unnecessary. Consider calling this `originalNames` or similar.

Reply:
calling it `originalItems`